src/entities.rs

```diff
@@ -224,6 +224,40 @@ impl Entities {
         }
     }
 
+    /// Allocate a specific entity ID and generation.  May return the location of the entity
+    /// previously allocated to the specified ID.
+    ///
+    /// Location should be written immediately.
+    pub fn alloc_at(&mut self, entity: Entity) -> Option<Location> {
+        self.verify_flushed();
+
+        let loc = if entity.id as usize >= self.meta.len() {
+            self.pending.extend((self.meta.len() as u32)..entity.id);
```

Comment:
`self.free_cursor` must be updated whenever `self.pending` changes size, or the next `flush` will malfunction.

Reply:
Should be fixed, please confirm.